src/Core/Framework/Test/Plugin/Command/PluginListCommandTest.php

```diff
@@ -0,0 +1,164 @@
+        static::assertStringEqualsFile(
+            __DIR__ . '/../_assertion/PluginListCommandTest::testCommand.txt',
+            implode("\n", array_map('trim', explode("\n", $output))) . "\n"
+        );
+    }
+
+    public function testFilter(): void
+    {
+        $filterValue = 'shopware-is-love';
+
+        $criteria = static::callback(function (Criteria $criteria) use ($filterValue): bool {
+            $filters = $criteria->getFilters();
+            // must be MultiFilter
+            if (!(\count($filters) === 1 && $filters[0] instanceof MultiFilter)) {
+                return false;
+            }
+            /** @var MultiFilter $filter */
+            $filter = $filters[0];
+            // must be OR
+            if ($filter->getOperator() !== MultiFilter::CONNECTION_OR) {
+                return false;
+            }
+            $fields = ['name', 'label'];
+            foreach ($filter->getQueries() as $query) {
+                /** @var ContainsFilter $query */
+                if (!(
+                    $query instanceof ContainsFilter
+                    && $query->getValue() === $filterValue
+                    // first test against name, then label
+                    && $query->getField() === array_shift($fields)
+                )
+                ) {
+                    return false;
+                }
+            }
+
+            return true;
+        });
+
+        $this->pluginRepoMock->method('search')->with($criteria, static::anything());
+
+        [$statusCode, $output] = $this->executeCommand(['--filter' => $filterValue]);
+        static::assertSame(0, $statusCode);
+        static::assertStringContainsString('Filtering for: ' . $filterValue, $output);
+    }
+
+    public function testJsonOutput(): void
+    {
+        $o1 = ['PLUGIN1'];
+        $o2 = ['PLUGIN2'];
+
+        $entities = [
+            $plugin1 = $this->createMock(PluginEntity::class),
+            $plugin2 = $this->createMock(PluginEntity::class),
+        ];
+
+        $plugin1->method('jsonSerialize')->willReturn($o1);
+        $plugin2->method('jsonSerialize')->willReturn($o2);
+
+        $this->setupEntityCollection($entities);
+
+        $options = ['--json' => true];
+        $json = json_encode([$o1, $o2], \JSON_THROW_ON_ERROR);
+        $expected = [0, $json];
+        static::assertSame($expected, $this->executeCommand($options));
+    }
+
+    private function executeCommand(array $options): array
+    {
+        $commandTester = new CommandTester($this->command);
+        $commandTester->execute($options);
+
+        return [$commandTester->getStatusCode(), trim($commandTester->getDisplay())];
+    }
+
+    private function setupEntityCollection(array $entities): void
+    {
+        $collection = $this->createMock(EntityCollection::class);
+        $collection->method('jsonSerialize')->willReturn($entities);
+        $collection->method('getIterator')->willReturn($this->arrayAsGenerator($entities));
```

Comment:
```suggestion
        $collection->method('getIterator')->willReturn(new ArrayIterator($entities));
```
That way you can get rid of the `arrayAsGenerator()` method ;)

Reply:
Unfortunately no :-/

Method getIterator may not return value of type ArrayIterator, its return declaration is "Generator"

Reply:
```php
    /**
     * @deprecated tag:v6.5.0 - Return type will be changed to \Traversable
     */
    #[\ReturnTypeWillChange]
    public function getIterator(): \Generator/* :\Traversable */
 ```
 
can you check if
```suggestion
        $collection->method('getIterator')->willReturn(yield from $entities);
```
may work?

Reply:
Nope :-/

```
Fatal error: Generators may only declare a return type of Generator, Iterator, Traversable, or iterable, void is not permitted in /var/www/html/platform/src/Core/Framework/Test/Plugin/Command/PluginListCommandTest.php on line 151
```

That yields from `setupEntityCollection` :-/